ov/server_hardware.go

```diff
@@ -208,6 +220,98 @@ func (s ServerHardware) GetPowerState() (PowerState, error) {
 	return pt.State, nil
 }
 
+// Add single rack server to the appliance
+func (c *OVClient) AddRackServer(rackServer ServerHardware) error {
+	log.Infof("Adding rack server %s.", rackServer.Hostname)
+	var (
+		uri = "/rest/server-hardware"
+		t   *Task
+	)
+	// refresh login
+	c.RefreshLogin()
+	c.SetAuthHeaderOptions(c.GetAuthHeaderMap())
+
+	t = t.NewProfileTask(c)
+	t.ResetTask()
+	log.Debugf("REST : %s \n %+v\n", uri, rackServer)
+	log.Debugf("task -> %+v", t)
+	data, err := c.RestAPICall(rest.POST, uri, rackServer)
+	if err != nil {
+		t.TaskIsDone = true
+		log.Errorf("Error submitting new rack server addition: %s", err)
+		return err
+	}
+
+	log.Debugf("Response New Rackserver %s", data)
+	if err := json.Unmarshal([]byte(data), &t); err != nil {
+		t.TaskIsDone = true
+		log.Errorf("Error with task un-marshal: %s", err)
+		return err
+	}
+
+	err = t.Wait()
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// Add multiple rack servers
+func (c *OVClient) AddMultipleRackServers(rackServer ServerHardware) error {
+	log.Infof("Adding multiple rack servers %s.", rackServer.Hostname)
+	var (
+		uri = "/rest/server-hardware/discovery"
+		t   *Task
+	)
+	// refresh login
+	c.RefreshLogin()
+	c.SetAuthHeaderOptions(c.GetAuthHeaderMap())
+
+	log.Debugf("REST : %s \n %+v\n", uri, rackServer)
+	log.Debugf("task -> %+v", t)
+	data, err := c.RestAPICall(rest.POST, uri, rackServer)
```

Comment:
Does this operation returns a task uri? If yes we need to check the task status for completion

Reply:
Done